From bf17e8ce7dc8b42afb09e1ba40efc438d6dd9a76 Mon Sep 17 00:00:00 2001
From: Chih-Wei Huang <cwhuang@linux.org.tw>
Date: Tue, 26 Feb 2019 11:27:33 +0800
Subject: [PATCH 7/9] android: avoid compile warnnings

To avoid the warnings(errors):

hardware/intel/common/vaapi/src/gen75_vme.c:1070:5: error: variable 'i965_kernel_num' is used uninitialized whenever switch default is taken [-Werror,-Wsometimes-uninitialized]
    default:
    ^~~~~~~
hardware/intel/common/vaapi/src/gen75_vme.c:1078:35: note: uninitialized use occurs here
    vme_context->vme_kernel_sum = i965_kernel_num;
                                  ^~~~~~~~~~~~~~~

hardware/intel/common/vaapi/src/gen9_vme.c:2036:5: error: variable 'i965_kernel_num' is used uninitialized whenever switch default is taken [-Werror,-Wsometimes-uninitialized]
    default:
    ^~~~~~~
hardware/intel/common/vaapi/src/gen9_vme.c:2044:35: note: uninitialized use occurs here
    vme_context->vme_kernel_sum = i965_kernel_num;
                                  ^~~~~~~~~~~~~~~

I am not going to disable the warning (by -Wno-sometimes-uninitialized)
since it's still a useful checking from the compiler.

Signed-off-by: Chih-Wei Huang <cwhuang@linux.org.tw>
---
 src/gen75_vme.c | 1 +
 src/gen9_vme.c  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/gen75_vme.c b/src/gen75_vme.c
index 64b7dc0..040dbf5 100644
--- a/src/gen75_vme.c
+++ b/src/gen75_vme.c
@@ -1070,6 +1070,7 @@ Bool gen75_vme_context_init(VADriverContextP ctx, struct intel_encoder_context *
     default:
         /* never get here */
         assert(0);
+        i965_kernel_num = 0;
 
         break;
     }
diff --git a/src/gen9_vme.c b/src/gen9_vme.c
index 19a9097..af7ef12 100644
--- a/src/gen9_vme.c
+++ b/src/gen9_vme.c
@@ -2036,6 +2036,7 @@ Bool gen9_vme_context_init(VADriverContextP ctx, struct intel_encoder_context *e
     default:
         /* never get here */
         assert(0);
+        i965_kernel_num = 0;
 
         break;
     }
-- 
2.20.1

